File: Kendaraan Practice/Bus.java
public class Bus extends Kendaraan{
    private int kapasitas;
    
    Bus(){
        super();
        this.kapasitas = 0;
    }

    Bus(int nomor, int tahun, String merk, int kapasitas){
        super(nomor, tahun, merk);
        this.kapasitas = kapasitas;
    }

    Bus(Bus other)
    {
        super(other);
        this.kapasitas = other.kapasitas;
    }

    public Bus copyUnit(Bus other){
        super.copyUnit(other);
        this.kapasitas = other.kapasitas;
        return this;
    }

    @Override
    public void printInfo()
    {
        super.printInfo();
        System.out.printf("Kategori: Bus%n");
        System.out.printf("Kapasitas: %d%n", this.kapasitas);
    }

    public int biayaSewa(int lamaSewa)
    {
        return (int)1e6*lamaSewa;
    }
}

File: Kendaraan Practice/Bus.cpp
#include<bits/stdc++.h>
#include "Bus.hpp"
#include "Kendaraan.hpp"

using namespace std;

// Default ctor
Bus::Bus() : Kendaraan()
{
    this->kapasitas = 0;
}

// User ctor
Bus::Bus(int nomor, int tahun, string merk, int kapasitas) : Kendaraan(nomor,tahun,merk)
{
    this->kapasitas = kapasitas;
}

// cctor
Bus::Bus(const Bus& other)
{
    this->nomor = other.nomor;
    this->tahun = other.tahun;
    this->merk = other.merk;
    this->kapasitas = other.kapasitas;
}

// dtor
Bus::~Bus() {}

void Bus::printInfo()
{
    Kendaraan::printInfo();
    printf("Kategori: Bus\n");
    printf("Cap\t: %d\n", this->kapasitas);
}

int Bus::biayaSewa(int lamaSewa)
{
    return 1e6*lamaSewa;
}